pkg/controllers/capr/autoscaler/autoscaler.go

```diff
@@ -0,0 +1,329 @@
+package autoscaler
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/lasso/pkg/dynamic"
+	"github.com/rancher/rancher/pkg/capr"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/generated/controllers/cluster.x-k8s.io/v1beta1"
+	fleetcontrollers "github.com/rancher/rancher/pkg/generated/controllers/fleet.cattle.io/v1alpha1"
+	mgmtcontrollers "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	v2provcontrollers "github.com/rancher/rancher/pkg/generated/controllers/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/wrangler"
+	wranglerv1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+type autoscalerHandler struct {
+	capiClusterCache           v1beta1.ClusterCache
+	capiMachineCache           v1beta1.MachineCache
+	capiMachineDeploymentCache v1beta1.MachineDeploymentCache
+
+	clusterClient v2provcontrollers.ClusterClient
+	clusterCache  v2provcontrollers.ClusterCache
+
+	globalRole      mgmtcontrollers.GlobalRoleClient
+	globalRoleCache mgmtcontrollers.GlobalRoleCache
+
+	globalRoleBinding      mgmtcontrollers.GlobalRoleBindingClient
+	globalRoleBindingCache mgmtcontrollers.GlobalRoleBindingCache
+
+	user      mgmtcontrollers.UserClient
+	userCache mgmtcontrollers.UserCache
+
+	token      mgmtcontrollers.TokenClient
+	tokenCache mgmtcontrollers.TokenCache
+
+	secret      wranglerv1.SecretController
+	secretCache wranglerv1.SecretCache
+
+	helmOp      fleetcontrollers.HelmOpController
+	helmOpCache fleetcontrollers.HelmOpCache
```

Comment:
Nit: naming
`secret wranglerv1.SecretController` is fine
`token      mgmtcontrollers.TokenClient` -> `tokenClient      mgmtcontrollers.TokenClient`
basically it's fine to omit a suffic for controllers, but sometimes the clients have `Client` and sometimes they don't.